server/src/main/java/org/elasticsearch/action/support/broadcast/node/BoundedDiagnosticRequestPermits.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.action.support.broadcast.node;
+
+import org.elasticsearch.common.settings.ClusterSettings;
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.util.concurrent.AdjustableSemaphore;
+
+/**
+ * This class guards the amount of bounded diagnostic requests a node can concurrently coordinate.
+ */
+public class BoundedDiagnosticRequestPermits {
+
+    public static final Setting<Integer> MAX_CONCURRENT_BOUNDED_DIAGNOSTIC_REQUESTS_PER_NODE = Setting.intSetting(
+        "cluster.max_concurrent_bounded_diagnostic_requests_per_node",
```

Comment:
Naming suggestion:

```suggestion
        "node.stats.max_concurrent_requests",
```

`max` and `bounded` are redundant, we mostly call them `stats` not `diagnostics`, and I think we can put this under `node.*` rather than calling it `cluster..._per_node`.

Reply:
> Sorry for the quick-fire reviews, this one is just alternative naming suggestions.

Thank you, this helps a lot to get the lingo right :)